tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java

```diff
@@ -151,6 +152,57 @@ public void testNegativeCasesInProjects(String projectName, String errorMsg) thr
+                {"invalid_module_structure", "invalid module structure found for module 'main'. " +
+                        "Please provide the module name as '[main]'" },
+                {"invalid_sub_module_structure", "invalid module structure found for module 'main.foo'. " +
+                        "Please provide the module name as '[main.foo]'" },
+                {"required_negative", "Value not provided for required configurable variable 'main:stringVar'"},
+                {"primitive_type_error", "configurable variable 'main:intVar' is expected to be of type 'int', " +
+                        "but found 'float'"},
+                {"primitive_structure_error", "configurable variable 'main:intVar' is expected to be of type 'int', " +
+                        "but found 'record'"},
+                {"array_type_error", "configurable variable 'main:intArr' is expected to be of type " +
+                        "'int[] & readonly', but found 'string'"},
+                {"array_structure_error", "configurable variable 'main:intArr' is expected to be of type " +
+                        "'int[] & readonly', but found 'record'"},
+                {"array_element_structure", "configurable variable 'main:intArr[2]' is expected to be of type 'int'," +
+                        " but found 'array'"},
+                {"array_multi_type", "configurable variable 'main:intArr[1]' is expected to be of type 'int'," +
+                        " but found 'string'"},
+                {"additional_field", "additional field 'scopes' provided for configurable variable 'main:testUser'" +
+                        " of record 'main:AuthInfo' is not supported"},
+                {"missing_record_field", "value not provided for non-defaultable required field 'username' of record" +
+                        " 'main:AuthInfo' in configurable variable 'main:testUser'"},
+                {"record_type_error", "configurable variable 'main:testUser' is expected to be of type " +
+                        "'main:(testOrg/main:0.1.0:AuthInfo & readonly)', but found 'string'"},
+                {"record_field_structure_error", "record field 'username' from configurable variable 'main:testUser' " +
+                        "is expected to be of type 'string', but found 'record'"},
+                {"record_field_type_error", "record field 'username' from configurable variable 'main:testUser' " +
+                        "is expected to be of type 'string', but found 'int'"},
+                {"missing_table_key", "value required for key 'username' of type 'table<(main:AuthInfo & readonly)>" +
+                        " key(username) & readonly' in configurable variable 'main:users'"},
+                {"table_type_error", "configurable variable 'main:users' is expected to be of type " +
+                        "'table<(main:AuthInfo & readonly)> key(username) & readonly', but found 'record'"},
+        };
+    }
+
```

Comment:
Shall we create an issue to support rest types in configurable?

Reply:
Included in #28976